File: python/bindings/micm/solver.cpp
// Copyright (C) 2023-2025 University Corporation for Atmospheric Research
// SPDX-License-Identifier: Apache-2.0
#include "../common.hpp"

#include <musica/micm/micm.hpp>
#include <musica/micm/micm_c_interface.hpp>

namespace py = pybind11;

void bind_micm_solver(py::module_ &m)
{
  py::enum_<musica::MICMSolver>(m, "_SolverType")
      .value("rosenbrock", musica::MICMSolver::Rosenbrock)
      .value("rosenbrock_standard_order", musica::MICMSolver::RosenbrockStandardOrder)
      .value("backward_euler", musica::MICMSolver::BackwardEuler)
      .value("backward_euler_standard_order", musica::MICMSolver::BackwardEulerStandardOrder)
      .value("cuda_rosenbrock", musica::MICMSolver::CudaRosenbrock);
}

File: julia/src/micm/solver.jl
# Copyright (C) 2023-2026 University Corporation for Atmospheric Research
# SPDX-License-Identifier: Apache-2.0

"""
    SolverType

Enum representing the type of MICM solver to use.

Values match the C++ `musica::MICMSolver` enum.
"""
@enum SolverType begin
    Rosenbrock = 1
    RosenbrockStandardOrder = 2
    BackwardEuler = 3
    BackwardEulerStandardOrder = 4
    CudaRosenbrock = 5
end
